compiler/rustc_codegen_llvm/src/back/archive.rs

```diff
@@ -208,7 +208,11 @@ impl<'a> ArchiveBuilder<'a> for LlvmArchiveBuilder<'a> {
         // have any \0 characters
         let import_name_vector: Vec<CString> = dll_imports
             .iter()
-            .map(|import| CString::new(import.name.to_string()).unwrap())
+            .map(if self.config.sess.target.arch == "x86" {
+                |import: &DllImport| CString::new(format!("_{}", import.name.to_string())).unwrap()
```

Comment:
This makes sense for functions with a `cdecl` calling convention, but `stdcall` [1] functions also have `@n` appended as part of the name mangling where n is the number of bytes in the parameter list. Is adding the underscore enough in that case?

[1] Not to mention `fastcall` and `vectorcall` which have their own decoration format.

Reply:
I don't expect it to be sufficient for stdcall, no, but I don't intend to support stdcall in this PR; that will be follow-on work.

Reply:
Yea, that's alright. Just add a FIXME pointing to #58713. I also updated it to note work is needed for different calling conventions.

Reply:
Would it be acceptable to add a diagnostic for stcall, fastcall, and vectorcall on i686, until I add support for those calling conventions?  Or do we need to add support for these as part of this PR?

Reply:
Sorry if I wasn't clear but yea, I'm fine with doing it in a followup PR. I also added another comment below to indicate where you might want to add the diagnostic. Just note this part with a `FIXME(#58713)` for the missing mangling work.

Besides, 32-bit windows is [clearly](https://docs.microsoft.com/en-au/windows-hardware/design/minimum/minimum-hardware-requirements-overview#31-processor) starting to be put on the backburner these days:
>Beginning with Windows 10, version 2004, all new Windows 10 systems will be required to use 64-bit builds and Microsoft will no longer release 32-bit builds for OEM distribution.